src/main/java/de/bonndan/nivio/assessment/AssessmentRepository.java

```diff
@@ -0,0 +1,48 @@
+package de.bonndan.nivio.assessment;
+
+import de.bonndan.nivio.model.FullyQualifiedIdentifier;
+import de.bonndan.nivio.model.Landscape;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+
+/**
+ * Stores and loads and creates Assessments with FullyQualifiedIdentifier as key
+ */
+@Component
+public class AssessmentRepository {
+
+    private HashMap<FullyQualifiedIdentifier, Assessment> repository;
+
+    public AssessmentRepository() {
+        repository = new HashMap<>();
+    }
+
+    public void clean() {
+        repository = new HashMap<>();
+    }
+
+    public Assessment createAssessment(@NonNull Landscape landscape) {
+        if (landscape != null) {
+            var newAssessment = AssessmentFactory.createAssessment(landscape);
+            storeAssessment(landscape.getFullyQualifiedIdentifier(), newAssessment);
+            return newAssessment;
+        } else {
+            return null;
+        }
+    }
+
+    public Assessment getAssessment(@NonNull FullyQualifiedIdentifier fullyQualifiedIdentifier) {
```

Comment:
If you use a @NonNull annotation you should ensure it is not null (e.g. use Objects.requireNonNull(...)) or instead annotate @Nullable or nothing. Declaring the parameter as final ensures the "pureness" of the method a bit.

The holy inquisition @Matthimatiker and @basseur might see that more strictly, though.  ✝️

Reply:
😉

Reply:
[![NOBODY EXPECTS THE INQUISITION](https://i.imgflip.com/3b32u8.jpg "NOBODY EXPECTS THE INQUISITION")](https://imgflip.com/i/3b32u8)

Reply:
🤣